src/components/Project/ProjectBasics/ProjectForm/ProjectHashTags/ProjectHashTags.tsx

```diff
@@ -152,21 +169,37 @@ const ProjectHashTagsDialog: FC<IProjectHashTagsDialogProps> = forwardRef(
     // Submit hashTagsForSubmit and close the dialog
     const onSubmit = useCallback(
       async (event: MouseEvent<HTMLButtonElement>) => {
-        try {
-          await patchProject({
-            id: projectId,
-            data: { hashTags: hashTagsForSubmit.map((h) => hashTagsObject[h.value].id) },
-          });
-          onChange(hashTagsForSubmit.map((h) => hashTagsObject[h.value].id));
-          toggleOpenDialog(event);
-        } catch (e) {
-          console.log('Error patching project hashtags: ', e);
-        }
+          try {
+            if (projectMode === 'new' && setHashTagsState) {
+              setHashTagsState((current) => ({
+                ...current,
+                projectHashTags: _.uniqWith(hashTagsForSubmit, _.isEqual)
+              }));
+            } else {
+              await patchProject({
+                id: projectId,
+                data: { hashTags: hashTagsForSubmit.map((h) => hashTagsObject[h.value].id) },
+              });
+            }
+            // Hashtags are saved to the form values in the onChange below
+            onChange(hashTagsForSubmit.map((h) => hashTagsObject[h.value].id));
+            toggleOpenDialog(event);
+          } catch (e) {
+            console.log('Error patching project hashtags: ', e);
+          }
       },
       [hashTagsForSubmit, projectId, toggleOpenDialog, onChange, hashTagsObject],
     );
 
     const handleClose = useCallback(() => {
+      if (projectMode === 'new') {
+        // If dialog is closed without saving set the initial values (values before making changes to the hashtags) back to the state
+       setHashTagsState((current) => ({
+          ...current,
+          projectHashTags: initialHashtagsForNewProject
+        }));
+      }
```

Comment:
Is this now needed either, cause state is only modified if the user saves the hashtags?